pynpoint/processing/psfsubtraction.py

```diff
@@ -189,67 +276,193 @@ def _run_multi_processing(self,
+                # 3.) median residuals
+                if self.m_res_median_out_port is not None:
+                    if self.m_processing_type in ['SDI+ADI', 'ADI+SDI']:
+                        out_array_medi[i, j] = combine_residuals(method='median',
+                                                                 res_rot=res_rot,
+                                                                 angles=parang)
+
+                    else:
+                        out_array_medi[i] = combine_residuals(method='median',
+                                                              res_rot=res_rot,
+                                                              angles=parang)
+
+                # 4.) noise-weighted residuals
+                if self.m_res_weighted_out_port is not None:
+                    if self.m_processing_type in ['SDI+ADI', 'ADI+SDI']:
+                        out_array_weig[i, j] = combine_residuals(method='weighted',
+                                                                 res_rot=res_rot,
+                                                                 residuals=residuals,
+                                                                 angles=parang)
+
+                    else:
+                        out_array_weig[i] = combine_residuals(method='weighted',
+                                                              res_rot=res_rot,
+                                                              residuals=residuals,
+                                                              angles=parang)
+
+                # 5.) clipped mean residuals
+                if self.m_res_rot_mean_clip_out_port is not None:
+                    if self.m_processing_type in ['SDI+ADI', 'ADI+SDI']:
+                        out_array_clip[i, j] = combine_residuals(method='clipped',
+                                                                 res_rot=res_rot,
+                                                                 angles=parang)
+
+                    else:
+                        out_array_clip[i] = combine_residuals(method='clipped',
+                                                              res_rot=res_rot,
+                                                              angles=parang)
 
-            # 5.) clipped mean residuals
-            if self.m_res_rot_mean_clip_out_port is not None:
-                stack = combine_residuals(method='clipped', res_rot=res_rot)
-                self.m_res_rot_mean_clip_out_port.append(stack, data_dim=3)
+        # Configurate data output according to the processing type
+        # 1.) derotated residuals
+        if self.m_res_arr_out_ports is not None and len(pca_first)+len(pca_secon) == 2 and self.m_processing_type != 'Oadi':
+            if pca_secon[0] == -1:
+                history = f'max PC number = {pca_first}'
+
+            else:
+                history = f'max PC number = {pca_first} / {pca_secon}'
+
+            squeezed = np.squeeze(out_array_res)
```

Comment:
What is happening here? Please add a comment.

Reply:
I added the squeezing to ensure uniform data output for the residuals. We discussed earlier that ADI+SDI and SDI+ADI should always produce 5 dimensional outputs. If you want the same for residuals I can add it but since the first two dimensions are always 1 this might not makes sense.

For now I added comments explaining this.

Reply:
Right, the current implementation seems fine, thanks.